include/nbl/video/ILogicalDevice.h

```diff
@@ -252,7 +252,25 @@ class NBL_API ILogicalDevice : public core::IReferenceCounted, public IDeviceMem
                 return nullptr;
             if (!layout->wasCreatedBy(this))
                 return nullptr;
-            return createDescriptorSet_impl(pool, std::move(layout));
+
+            uint32_t descriptorCount[asset::EDT_COUNT] = { 0 };
+            for (const auto& binding : layout->getBindings())
+                descriptorCount[binding.type] += binding.count;
```

Comment:
this is why we should precompute these counts and store them in `IDescriptorSetLayout`

Reply:
Resolved.